src/components/Tabs/TabItem.tsx

```diff
@@ -13,27 +14,28 @@ export type TabItemProps = {
     decorator?: ReactElement;
     badge?: BadgeProps;
     children: ReactNode;
+    active?: boolean;
 };
 
-export const TabItem = ({ active, disabled, children, id }: TabItemProps & { active?: boolean }): ReactElement => {
-    const ref = useRef<HTMLDivElement | null>(null);
+export const TabItem = ({ active, disabled, children, id }: TabItemProps): Nullable<ReactElement> => {
+    const [ref, setRef] = useState<HTMLDivElement | null>(null);
     const [hasInteractiveElements, setHasInteractiveElements] = useState(false);
 
     useEffect(() => {
-        setHasInteractiveElements(checkIfContainInteractiveElements(ref.current));
-    }, [children]);
+        setHasInteractiveElements(checkIfContainInteractiveElements(ref));
+    }, [ref]);
 
-    return (
+    return active ? (
         <div
             role="tabpanel"
-            ref={ref}
+            ref={setRef}
             id={`${id}-content`}
             aria-labelledby={id}
-            className={merge([!active || disabled ? 'tw-hidden' : '', 'focus:tw-ring-4 focus:tw-ring-blue'])}
+            className={merge([!active || disabled ? 'tw-hidden' : '', !hasInteractiveElements && FOCUS_VISIBLE_STYLE])}
```

Comment:
Sorry, in the snippet I sent you, my branch wasn't updated yet with the latest changes I made on the Tab.
So the 'focus:tw-ring-4 focus:tw-ring-blue' needs to stay instead of the !hasInteractiveElements && FOCUS_VISIBLE_STYLE

Reply:
Updated. can you check, please?

Reply:
All good with the focus style, thanks.